src/main/java/org/jabref/logic/database/DuplicateCheck.java

```diff
@@ -101,21 +103,24 @@ private static boolean haveDifferentChaptersOrPagesOfTheSameBook(final BibEntry
 
     private static double[] compareRequiredFields(final BibEntryType type, final BibEntry one, final BibEntry two) {
         final Set<OrFields> requiredFields = type.getRequiredFields();
-        return requiredFields == null
+        return requiredFields.isEmpty()
                 ? new double[] {0., 0.}
```

Comment:
This could still lead to an npe now or is ensured that the field set is always initialized?

Reply:
Forget it. I checked, it's always initialized.